### ISSUE TITLE
GameGuru MAX - Enable Grid Mode - Should Be Able to Enable BEFORE Placing Entities

### ISSUE THREAD

**Issue by ArgentArts** [TheGameCreators/GameGuruRepo#1968]
Currently, if you have nothing selected in MAX, you cannot get to and Enable Grid Mode. This means you need to first select something and go to place it in the viewport BEFORE you can enable it, even if other things are currently in the viewport (if you've had it turned off previously). A setting like Enable Grid Mode should not be dependent on having anything selected at all, but should be something that  can be accessed and turned on/off at any time. When I first start with a blank level, I want to start by setting up my snapping, then start to place my entities, especially for an indoor only level. 

Please make Enable Grid Mode available, even if nothing is selected so we can turn it on/off even if nothing is selected.

**Comment by Necrym59** [comment]
Agreed it should be a on/off keystroke, also it would be nice to have a view of the grid in the editor view options to i think.

**Comment by ArgentArts** [comment]
@LeeBamberTGC @dllllllllllb I really want to bump this because I think it's needed. It's very strange to have to place an object FIRSST before setting up your grid, especially if you're working with an interior level that is built with modular pieces. If, for example, I am starting a new day's work on a project I had left off of from the previous day, I already have pieces set in place. But MAX will start up with no grid snapping and I can't just set it. I have to first select something or place a new object before I gain access to the ability to set snapping. 

It's pretty standard to be able to set up your work environment before starting your work. This includes setting up snapping. Please consider making snapping something that always shows up, even if nothing is selected or being placed.

**Comment by LeeBamberTGC** [comment]
@rvanner I had the same feeling. It seems daft the grid only shows when an object is selected. We can leave it visible even if no objects are selected (will sit above the Tutorial and Keyboard Shortcuts components when in object mode).  This way grids can be set up before any work starts. Thoughts?

**Comment by rvanner** [comment]
@LeeBamberTGC About 6 months ago I argued to the grid to be on it's own. Glad you've come round to the idea finally! :-)

**Comment by ArgentArts** [comment]
Snapping, to me, is very important, especially when working on an interior level. As a result, I might actually suggest something like this:

![snap](https://user-images.githubusercontent.com/36735507/165356099-0930abdc-4ffb-4355-8a87-11636c13314d.png)

There's plenty of space on that top bar with nothing going on in the middle. Why not use that empty space?

**Comment by LeeBamberTGC** [comment]
@ArgentArts It is all about the smart use of real estate UI. Grid has no influence on terrain mode and not related to test game. That top bar is to select the top tier options for moving through the level editor.  For consistency it would be on the right panel somewhere.

**Comment by ArgentArts** [comment]
Fair enough. :)
